db/types/money.py

```diff
@@ -25,3 +29,18 @@ def install(engine):
         conn.execute(text(drop_domain_query))
         conn.execute(text(create_domain_query))
         conn.commit()
+
+
+def get_money_array(table_oid, engine, column_attnum):
+    table = reflect_table_from_oid(table_oid, engine)
+    column_name = get_column_name_from_attnum(table_oid, column_attnum, engine)
+    with engine.begin() as conn:
+        package_func = getattr(func, get_qualifier_prefix())
+        money_func = getattr(package_func, MONEY_ARR_FUNC_NAME)
+        money_select = money_func((table.c[column_name]), type_=ARRAY(String))
+        sel = select(money_select).where(money_select[4] is not None)
+        data = conn.execute(sel).fetchone()
```

Comment:
Did you double-check that this doesn't run the function for all rows before returning one in `fetchone`? The function in question is pretty intense, and I think it might make sense to limit the query (unless it's not actually running for all rows).

Reply:
Nice catch! Got accustomed to Django's behaviour and forgot we are dealing with cursors here.